examples/typescript-jest-node-fetch/src/github-api.test.ts

```diff
@@ -0,0 +1,31 @@
+/** @jest-environment setup-polly-jest/jest-environment-node */
+import autoSetupPolly from './utils/auto-setup-polly';
+import { getUser } from './github-api';
+
+describe('github-api client', () => {
+  let pollyContext = autoSetupPolly();
```

Comment:
I'm testing this library in our test suite. My feeling is that Polly should be as automatic as possible and devs shouldn't have to think about it (unless they want to mock a request). I found that I could set up Polly without having to add anything manually to tests using this:

``` typescript
// tests/setupTestsAfterEnv.ts

import path from 'path';
import { setupPolly } from 'setup-polly-jest';
import NodeHttpAdapter from '@pollyjs/adapter-node-http';
import FSPersister from '@pollyjs/persister-fs';

global.pollyContext = setupPolly({
  adapters: [NodeHttpAdapter],
  persister: FSPersister,
  persisterOptions: {
    fs: {
      recordingsDir: path.resolve(__dirname, '__recordings__'),
    },
  },
  // logLevel: 'info',
  recordIfMissing: false,
  recordFailedRequests: true,
  expiresIn: '30d',
});

// tests/global.d.ts
// note that you will have to include this in your tsconfig.json somehow (automatically or manually)

import { setupPolly } from 'setup-polly-jest';

declare global {
  /* eslint-disable-next-line @typescript-eslint/no-namespace */
  namespace NodeJS {
    interface Global {
      pollyContext: ReturnType<typeof setupPolly>;
    }
  }
}

// jest.config.js
module.exports = {
  // ...
  setupFilesAfterEnv: ['./tests/setupTestsAfterEnv.ts'],
  // ...
}
```

This way `pollyContext` is magically available in any test.

What do you think about this approach?

Reply:
I love where you're going @mcmire - I think Polly is 🔥 when it's transparent, so your code doesn't change depending on if you want to mock or not.

I was going to include something similar to this pattern! The reason I didn't is because I think an opt-in design can meet more use cases.
As it still allows you to wire it up in `setupFilesAfterEnv` scripts:

```ts
// `./tests/setupFilesAfterEnv.ts`
import { autoSetupPolly } from './src/utils/auto-setup-polly';

declare global {
  namespace NodeJS {
    interface Global {
      pollyContext: ReturnType<typeof autoSetupPolly>;
    }
  }
}

global.pollyContext = autoSetupPolly();

export default global.pollyContext;
```

Maybe another example or some comments in the README is warranted?

Reply:
Ah yeah that's a fair point. This technique is also useful if you're slowly introducing Polly into a test suite, I'd imagine. In that case, an extra example would be nice, but not necessary.